pkg/sql/execinfrapb/processors_bulk_io.proto

```diff
@@ -511,7 +517,7 @@ message LogicalReplicationWriterSpec {
     // Checkpoint stores a set of resolved spans denoting completed progress.
     optional jobs.jobspb.StreamIngestionCheckpoint checkpoint = 7 [(gogoproto.nullable) = false];
 
-    // TableDescriptors is a map from destination table IDs to the source table
-    // descriptors.
-    map<int32, cockroach.sql.sqlbase.TableDescriptor> table_descriptors = 8 [(gogoproto.nullable) = false];
+    // TableReplicationMetadata is a set of metadata containing destination table ID,
+    // source table descriptor, and database prefix for each table to be replicated.
+    repeated TableReplicationMetadata table_metadata = 8 [(gogoproto.nullable) = false];
```

Comment:
why did you convert this into a slice? I think what steven and I were envisioning was:

```
type TableMetdata struct {
  sourcDescriptor descriptor
  databaseName string
}
```

then we keep this map here:
```
map[tableID]TableMetadata
```